src/main/java/frc/robot/utils/swerve/SwerveSetpointGenerator.java

```diff
@@ -0,0 +1,395 @@
+package frc.robot.utils.swerve;
+
+import static frc.robot.utils.math.EqualsUtil.*;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.geometry.Twist2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import frc.robot.utils.math.EqualsUtil;
+import frc.robot.utils.math.GeomUtil;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import lombok.Builder;
```

Comment:
This file doesn't compile for me.  It throws errors with the lombok stuff, and then even if that is commented out, it can't find ChassisSpeeds.toTwist2d()  Will not compile under WPILib 2024.
https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/math/kinematics/ChassisSpeeds.html
Where did this file come from?

Reply:
This is 6328's swerve setpoint generator, which uses the velocity and constraint limits to find the best kinematic solution for the module states. I will link the source chief Delphi post. It was not bugging out on me at all, so I can't comment on that.